enhancements/monitoring/optional-monitoring-capability.md

```diff
@@ -0,0 +1,967 @@
+including any managed manifests, so that I can minimize the resource
+consumption of monitoring on my cluster.
+> As a cluster administrator, I want to be able to disable as much
+of the in-cluster monitoring stack as possible right from the
+install-time, so I don't have to manually configure the in-cluster
+monitoring stack, as well as other components that depend on it,
+after the cluster is up and running.
+
+### Goals
+
+<!--
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+-->
+
+* Allow reducing the monitoring footprint on clusters that do not
+  require a full-fledged monitoring solution, while ensuring that
+  telemetry operations are not affected (admins can still disable
+  telemetry if they want to).
+* Teach other components about the monitoring capability, so they
+  can adjust their behavior accordingly (e.g., CVO, hypershift).
+* Introduce the ability to harness this feature at install-time,
+  so that admins don't have to manually configure the in-cluster
+  monitoring stack, as well as other components that depend on it,
+  after the cluster is up and running. This pattern additionally
+  allows for more components to be "taught" about the capability
+  to become more resource-aware from a monitoring perspective from
+  the get-go.
+* Allow other components to have a clear signal about whether there's
+  a monitoring stack deployed, by building over cluster capabilities,
+  rather than relying on heuristics (e.g., checking for the presence
+  of certain resources).
+* Preserve the historical UX by enabling the capability by default
+  (implicitly enabled).
+* Ensure that disabling the capability does not affect telemetry
+  operations.
+* Owing to the reduced volume of monitoring data collected, move
+  away from the HA Prometheus configuration to a single replica
+  configuration, when the capability is disabled.
+* Disable all optional components of the in-cluster monitoring stack, 
+  when the capability is disabled. This includes:
+  - Alertmanager
+  - Thanos Ruler
+  - Thanos Querier
+  - The User Workload Monitoring Stack
+* Expose the capability's status through logs, an info metric, as
+  well as `ClusterOperator`'s status.
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+
+* Monitoring components that are not required for telemetry operations,
+  but still valuable for monitoring use-cases, will not be disabled
+  when the capability is disabled. For example, metrics-server will
+  remain enabled to ensure that the autoscaling pipelines are not
+  affected.
+* Monitoring components that other capabilities rely on, and disabling
```

Comment:
I'm not sure about this point. Do we say that monitoring-plugin will be deployed?